### PR TITLE
fix(general-petition): Removing unused feature flags

### PR DESCRIPTION
**Pull request by albinagu**
## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by kksteini — APPROVED** [review]
Nice! :+1:

File: libs/application/templates/driving-license-duplicate/src/lib/drivingLicenseDuplicateTemplate.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { Events, States, Roles } from './constants'
 import { dataSchema } from './dataSchema'
 import { m } from './messages'
 import { ApiActions } from './constants'
-import { FeatureFlagClient, Features } from '@island.is/feature-flags'
+import { FeatureFlagClient } from '@island.is/feature-flags'
 import {
   DrivingLicenseDuplicateFeatureFlags,
   getApplicationFeatureFlags,
```
Comment (saevarma) [inline libs/application/templates/driving-license-duplicate/src/lib/drivingLicenseDuplicateTemplate.ts:24]: What about this?

And also L#76-87?

File: libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts [hunk 2]
```diff
@@ -16,7 +16,6 @@ import {
   DistrictsApi,
 } from '@island.is/application/types'
 import { assign } from 'xstate'
-import { Features } from '@island.is/feature-flags'
 import { getSpouseNationalId } from './utils'
 import { FeatureFlagClient } from '@island.is/feature-flags'
```
Comment (saevarma) [inline libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:20]: What about this and L#60-70 ?

File: libs/application/templates/operating-license/src/lib/OperatingLicenseTemplate.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ import {
 import { dataSchema } from './dataSchema'
 import { Roles, States, Events, ApiActions } from './constants'
 import { m } from './messages'
-import { FeatureFlagClient, Features } from '@island.is/feature-flags'
+import { FeatureFlagClient } from '@island.is/feature-flags'
```
Comment (saevarma) [inline libs/application/templates/operating-license/src/lib/OperatingLicenseTemplate.ts:15]: Ditto and the formloader below?

File: libs/application/templates/driving-license-duplicate/src/lib/drivingLicenseDuplicateTemplate.ts (hunk 1 above)
Comment (albinagu) [reply libs/application/templates/driving-license-duplicate/src/lib/drivingLicenseDuplicateTemplate.ts:24]: these are being used for testing on dev with fake data

File: libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts (hunk 2 above)
Comment (albinagu) [reply libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:20]: keeping this for now, 
for testing on dev with fake data

File: libs/application/templates/operating-license/src/lib/OperatingLicenseTemplate.ts (hunk 3 above)
Comment (albinagu) [reply libs/application/templates/operating-license/src/lib/OperatingLicenseTemplate.ts:15]: same as above

File: libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts (hunk 2 above)
Comment (saevarma) [reply libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:20]: Okay, I see. Fake Data sounds like something that should always be an option in the dev environment and I feel like the feature flags is not the correct tool to use for that. You should look into the parental leave application, that renders on dev a prerequisite subsection that asks the user if he wants to use fake data or not.

**Review by saevarma — APPROVED** [review]
LGTM and nice clean up on feature flags.

But check out my comment for possible improvements on the fake data and further clean up on feature flags.

File: libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts (hunk 2 above)
Comment (albinagu) [reply libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:20]: @saevarma 

Okay, parental leave application, however, is using the deprecated `isRunningOnEnvironment` check. The comment above it says "Any decisions based on the environment name should be rather based on feature flags and/or configuration." 

I would also want to be able to clean up feature flags thats are used for fake data, but im rather unsure at this point what the best way is. Perhaps good food for thought, to establish what we want to continue with.

Comment (saevarma) [reply libs/application/templates/marriage-conditions/src/lib/MarriageConditionsTemplate.ts:20]: I missed that comment. Thanks for clarifying. I will take this up with Norda about general strategy on fake data in dev environment.

**Pull request merged**